packages/client/hmi-client/src/components/workflow/tera-simulate-chart.vue

```diff
@@ -228,8 +231,47 @@ const renderGraph = () => {
 					borderWidth: lineWidthArray.value[runIdx]
 				};
 				datasets.push(dataset);
-			})
-	);
+			});
+
+		if (props.initialData) {
+			let v = variable;
+			// get the dataset variable from the mapping of model variable to dataset variable if there's a mapping
+			if (props.mapping) {
+				if (!(props.mapping.length === 1 && Object.values(props.mapping[0]).some((val) => !val))) {
+					const varMap = props.mapping.find((m) => m.modelVariable === variable);
+					if (varMap) {
+						v = varMap.datasetVariable;
+					}
+				}
+			}
+
+			// get  the index of the timestamp column
+			const tIndex = props.initialData.headers.indexOf('timestamp');
+
+			// get the index of the variable column
+			let colIdx: number;
+			for (let i = 0; i < props.initialData.headers.length; i++) {
+				if (props.initialData.headers[i].trim() === v.trim()) {
+					colIdx = i;
+					break;
+				}
+			}
+
+			const dataset = {
+				// ignore the first row, it's the header
+				data: props.initialData.csv.slice(1).map((datum: string[]) => ({
+					x: +datum[tIndex].trim(),
+					y: +datum[colIdx].trim()
```

Comment:
I don't think you need `trim()` with coercion ?

```
> let x = "   22.123  "
< undefined
> x
< '   22.123  '
> +x
< 22.123
```

Reply:
Oh, does it automatically get rid of whitespaces and newlines and such? I initially added this trim because some of the values were like `"123.456\r"` so I wanted to make sure we got rid of the `\r`. I didn't realize this was redundant. I will get rid of the `trim()` then.

Reply:
Ugh...if we are getting `\r` or `\n` or some combination of them, it means we didn't parse the file properly 😢